File: Java/Matrix Validation/Matrix validation.java
class matrix_validation {
    public static void main(String[] args) {
        int[][] A = {
            { 1, 0, 0 },
            { 0, 1, 0 },
            { 0, 0, 1 }
        };
        
        boolean flag = true; int sameCount = 0, zeroCount = 0;  
        System.out.println("Matrix: ");
        for (int row = 0; row < A.length; row++) {
            for (int col = 0; col < A[row].length; col++) {
                System.out.print("\t" + A[row][col]);
                if((row == col && row != 0 && col != 0) && (A[row-1][col-1] == A[row][col])){
                    sameCount++;
                }else if(row != col && A[row][col] != 0){
                    flag = false;
                }
                if (A[row][col] == 0) {
                    zeroCount++;
                }
            }
            System.out.println();
        }
        
        System.out.println();
        if (zeroCount == (A.length * A[0].length)) {
            System.out.println("It's a Null matrix");
        }else if(A.length == A[0].length){
            if(flag == true){
                if((sameCount+1) == A.length){
                    if(A[0][0] == 1){
                        System.out.println("It's a Identitiy matrix");
                    }else{
                        System.out.println("It's a Scaler matrix");
                    }
                }else{
                    System.out.println("It's a Diagonal matrix");
                }
            }else{
                System.out.println("It's not a Diagonal neither Identity, Scaler or Null matrix. \nBut it's a Square matrix.");
            }
        }else if(A.length == 1 && A[0].length > 1){
            System.out.println("It's a Row matrix.");
        }else if(A.length > 1 && A[0].length == 1){
            System.out.println("It's a Column matrix.");
        }else{
            System.out.println("It's a Rectangle matrix.");
        }
    }
}

File: C++/Matrix Validation/Matrix validation.cpp
#include <iostream>
#include <vector>
using namespace std;

int main(){
    vector <vector<int>> A = {
        { 1, 0, 0 },
        { 0, 1, 0 },
        { 0, 0, 1 }
    };
    
    bool flag = true; int sameCount = 0, zeroCount = 0;  
    cout << "Matrix: " << endl;
    for (int row = 0; row < A.size(); row++) {
        for (int col = 0; col < A[row].size(); col++) {
            cout << "\t" << A[row][col];
            if((row == col && row != 0 && col != 0) && (A[row-1][col-1] == A[row][col])){
                sameCount++;
            }else if(row != col && A[row][col] != 0){
                flag = false;
            }
            if (A[row][col] == 0) {
                zeroCount++;
            }
        }
        cout << endl;
    }
    
    cout << endl;
    if (zeroCount == (A.size() * A[0].size())) {
        cout << "It's a Null matrix" << endl;
    }else if(A.size() == A[0].size()){
        if(flag == true){
            if((sameCount+1) == A.size()){
                if(A[0][0] == 1){
                    cout << "It's a Identitiy matrix" << endl;
                }else{
                    cout << "It's a Scaler matrix" << endl;
                }
            }else{
                cout << "It's a Diagonal matrix" << endl;
            }
        }else{
            cout << "It's not a Diagonal neither Identity, Scaler or Null matrix. \nBut it's a Square matrix." << endl;
        }
    }else if(A.size() == 1 && A[0].size() > 1){
        cout << "It's a Row matrix." << endl;
    }else if(A.size() > 1 && A[0].size() == 1){
        cout << "It's a Column matrix." << endl;
    }else{
        cout << "It's a Rectangle matrix." << endl;
    }
}